examples/discovery-search-app/src/App.js

```diff
@@ -28,9 +28,48 @@ const App = () => {
     version: '2019-01-01',
     authenticator
   });
+  const [projectId, setProjectId] = useState(process.env.REACT_APP_PROJECT_ID);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
 
-  return (
-    <DiscoverySearch searchClient={searchClient} projectId={process.env.REACT_APP_PROJECT_ID}>
+  if (process.env.REACT_APP_CYPRESS_MODE && !projectId) {
+    console.warn('Running in feature test mode, setting REACT_APP_PROJECT_ID to "cypress"');
+    setProjectId('cypress');
+  }
+
+  useEffect(() => {
+    async function fetchProjects() {
+      try {
+        const {
+          result: { projects }
+        } = await searchClient.listProjects();
+        setProjectId(
+          projects.find(project => project.name === 'Sample Project')?.project_id ||
+            projects[0]?.project_id ||
+            'NOT_FOUND'
```

Comment:
I'm curious if setting this to a non-null value for the case where a project id is not found has any repercussions on what gets displayed. It looks like our default params in `DiscoverySearch.tsx` all use `''` for the project id.

Reply:
the reason i liked this approach is because its more clear that no projects were found rather than some sort of programming error (i.e. something returned back an empty string that wasn't supposed to). empty string is a useful default but also triggers a lot of `!value` expressions in javascript (since empty string is falsy)

Reply:
I can get on board with that. Still, maybe we should add a warning for `projectId === 'NOT_FOUND'`, similar to [the warning below](https://github.com/watson-developer-cloud/discovery-components/pull/180/files#diff-519cf620a75f8f39ac471a84a7015fd95b3eeed9a603ad5aadddabcdc7fde664R59-R61), so that the user or dev knows that there's still not a valid project id, even if the example app starts up successfully. Digging for the project id in order to discover that it wasn't found seems like extra debugging work we could save a user from.